app/controllers/api/motorcycles_controller.rb

```diff
@@ -1,2 +1,44 @@
 class Api::MotorcyclesController < ApplicationController
+  before_action :set_motorcycle, only: [:show, :update, :destroy]
+
+  def index
+    @motorcycles = Motorcycle.all
+    render json: @motorcycles
+  end
+
+  def show
+    render json: @motorcycle
+  end
+
+  def create
+    @motorcycle = Motorcycle.new(motorcycle_params)
+
+    if @motorcycle.save
+      render json: @motorcycle, status: :created, location: @motorcycle
+    else
+      render json: @motorcycle.errors, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @motorcycle.update(motorcycle_params)
+      render json: @motorcycle
+    else
+      render json: @motorcycle.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @motorcycle.destroy
+    head :no_content
+  end
```

Comment:
- First we need to check if motor exists, if not we will return a bad requset
- then after delete send a `status code of 200`

Reply:
Added the check condition.